### PR TITLE
move cgdfg to a module in dfg

### PR DESCRIPTION
**Pull request by Affie**
~~- [ ] still needs some cleanup~~ Going to do this in separate PR with test improvements

### CODE REVIEW

File: src/CloudGraphsDFG/CloudGraphsDFG.jl [hunk 1]
```diff
@@ -1,6 +1,94 @@
...
+                                    getPPE,
+                                    addPPE!,
+                                    listPPEs,
+                                    deletePPE!,
+                                    updatePPE!,
+                                    packPPE,
+                                    unpackPPE,
+                                    getSolvable,
+                                    setSolvable!
+
```
Comment (Affie) [inline src/CloudGraphsDFG/CloudGraphsDFG.jl:76]: Still need to confirm if everything is here.

File: src/DistributedFactorGraphs.jl [hunk 2]
```diff
@@ -273,24 +274,28 @@ include("services/CompareUtils.jl")
...
         @info "Including Plots"
         include("DFGPlots/DFGPlots.jl")
         @reexport using .DFGPlots
     end
 
+    if get(ENV, "DFG_USE_CGDFG", "") == "true"
+        @info "Including CloudGraphsDFG"
+        @eval Base.include(DistributedFactorGraphs, joinpath(@__DIR__, "CloudGraphsDFG/CloudGraphsDFG.jl"))
+        @eval DistributedFactorGraphs @reexport using .CloudGraphsDFGs
+    end
```
Comment (Affie) [inline src/DistributedFactorGraphs.jl:297]: @dehann How do we want to do this?
The other way is to do the check at compile time, but then you have to manually trigger the compile if you want to change it.

Comment (dehann) [reply src/DistributedFactorGraphs.jl:297]: Not sure i understand the other way...?  But either way ENV variable is good/required, and its also fine to trigger new precompile if thats needed.

**Comment by dehann** [comment]
looks good thanks @Affie

File: src/DistributedFactorGraphs.jl (hunk 2 above)
Comment (Affie) [reply src/DistributedFactorGraphs.jl:297]: Other way is not in `function __init__()`, so its pre-compiled.

**Pull request merged**